code/go/pkg/validator/validator_fields_test.go

```diff
@@ -0,0 +1,521 @@
+			},
+		},
+		{
+			title:           "array",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name": "afield",
+					"type": "array",
+				},
+			},
+			expectedErrors: []string{
+				`field 0.type: 0.type must be one of the following: "aggregate_metric_double", "alias", "histogram", "constant_keyword", "text", "match_only_text", "keyword", "long", "integer", "short", "byte", "double", "float", "half_float", "scaled_float", "date", "date_nanos", "boolean", "binary", "integer_range", "float_range", "long_range", "double_range", "date_range", "ip_range", "group", "geo_point", "object", "ip", "nested", "flattened", "wildcard", "version", "unsigned_long"`,
+			},
+		},
+		{
+			title:           "invalid custom date",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":        "my_custom_date",
+					"type":        "keyword",
+					"date_format": "yyy-MM-dd",
+				},
+			},
+			expectedErrors: []string{
+				`field "my_custom_date" of type keyword can't set date_format. date_format is allowed for date field type only`,
+			},
+		},
+		{
+			title:           "required object type",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name": "object_without_object_type.*",
+					"type": "object",
+				},
+			},
+			expectedErrors: []string{
+				`field 0: object_type is required`,
+			},
+		},
+		{
+			title:           "object with subfields should be of type group",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":        "object_with_subfields",
+					"type":        "object",
+					"object_type": "keyword",
+					"fields": []map[string]any{
+						{
+							"name": "foo",
+							"type": "keyword",
+						},
+					},
+				},
+			},
+			expectedErrors: []string{
+				`field 0.type: 0.type must be one of the following: "group", "nested"`,
+			},
+		},
+		{
+
+			title:           "enabled object without object type",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":    "someobject",
+					"type":    "object",
+					"enabled": true,
+				},
+			},
+			expectedErrors: []string{
+				`field 0.enabled: 0.enabled does not match: false`,
+			},
+		},
+		{
+
+			title:           "disabled object with object type",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":        "someobject",
+					"type":        "object",
+					"object_type": "keyword",
+					"enabled":     false,
+				},
+			},
+			expectedErrors: []string{
+				`field 0.enabled: 0.enabled does not match: true`,
+			},
+		},
+		{
+			title:           "bad disabled object was allowed till 3.0.2",
+			packageTemplate: "integration_v3_0_2",
```

Comment:
Could it be added a `version` field to set the `format_version` in the manifest.yml when creating the temporal package for this unit test?

I see for instance that there are some tests using `integration_v3_0`  and another `integration_v3_0_2` templates. Looking into these templates, they look the same package just changing the spec version.

That would allow us to create a template just per major (o minor?), or just packages containing some specific settings like the ones for time series.
WDYT?

Reply:
I thought about this, but there are versions that require more changes apart from the format version in the manifest. It is true that this would help though in this case. I will give it a try. We will still need to keep the `packageTemplate`.